src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -888,22 +888,22 @@ namespace cryptonote
 
 
     {
-      m_core.pause_mine();
-      epee::misc_utils::auto_scope_leave_caller scope_exit_handler = epee::misc_utils::create_scope_leave_handler(
-        boost::bind(&t_core::resume_mine, &m_core));
-
       MLOG_YELLOW(el::Level::Debug, "Got NEW BLOCKS inside of " << __FUNCTION__ << ": size: " << arg.blocks.size());
 
       if (m_core.get_test_drop_download() && m_core.get_test_drop_download_height()) { // DISCARD BLOCKS for testing
 
-        uint64_t previous_height = m_core.get_current_blockchain_height();
-
         // we lock all the rest to avoid having multiple connections redo a lot
         // of the same work, and one of them doing it for nothing: subsequent
         // connections will wait until the current one's added its blocks, then
         // will add any extra it has, if any
         CRITICAL_REGION_LOCAL(m_sync_lock);
 
+        m_core.pause_mine();
```

Comment:
The mining threads are still alive when paused, they just don't start another hash in the loop if they see a pause is requested. So, technically, mining may still be going on at the time pause_mining() returns, for a duration of one hash.
I might add a cond var to that code, since the restart is delayed by 100ms or so for the same polling reason.

But that's not what I had meant in my earlier comment: I'd meant that pausing the mining *after* the lock is taken would have a material effect only if the lock too a while being acquired. And that can only happen if that same lock is already held by another thread, and it's the only place which uses that lock, so it means that that other thread has already paused mining anyway.